docs/source/usage.rst

```diff
@@ -70,156 +69,182 @@ generates a STAC Item with this id.
                properties={},
          )
 
+         self.logger.debug(f"Created Item with id '{item.id}'")
+
          return [item.to_dict()]
 
    # Support for running as a Lambda Function
    def handler(event: dict[str, Any], context: Any) -> dict[str, Any]:
-      return MyTask.handler(event)  # type: ignore
+      return MyTask.handler(event)
 
    # Support for running as a CLI application
    if __name__ == "__main__":
       MyTask.cli()
 
-The expected input looks like the following:
+A possible input payload looks like the following:
 
 .. code-block:: json
 
    {
       "id": "my-task/workflow-my-task/5427299e5b635537f33c07e0ad32fb87",
-      "item_id": "G23923",
-      "process": {
-         "upload_options": {
-            "collections": {
-               "my-collection": "$[?(@.id =~ '.*')]"
+      "process": [
+         {
+            "collection_matchers": [
+               {
+                  "type": "jsonpath",
+                  "pattern": "$[?(@.id =~ '.*')]",
+                  "collection_name": "my-collection"
+               }
+            ],
+            "collection_options": {
+               "my-collection": {
+                  "upload_options": {
+                     "path_template": "{collection}/{year}/{month}/{day}/{item_id}"
+                  }
+               }
```

Comment:
I probably should have called this out before, but I view collection_options as a bit of an "advanced" feature. Consider most users shouldn't need anything more than `upload_options`. As such, I'm not sure it makes sense to include this in a basic example. Or, if you do, I'm not sure it makes sense without another case that falls back to `upload_options` to show that you can still use `upload_options`.

Reply:
Hmm. True. In most cases, the existing single upload option is OK. I'll take a look.

Reply:
Changed to show and discuss the "global" `upload_options` instead.